torch/csrc/jit/codegen/cuda/iter_visitor.cpp

```diff
@@ -26,9 +28,37 @@ std::vector<Statement*> IterVisitor::next(Val* v) {
   return {};
 }
 
-std::vector<Statement*> IterVisitor::next(Expr* expr) {
+std::vector<Statement*> IterVisitor::next(Expr* expr, bool respect_compute_at) {
   FusionGuard::getCurFusion()->assertInFusion(expr, "Cannot traverse expr, ");
-  return {expr->inputs().begin(), expr->inputs().end()};
+  std::vector<Statement*> next_stmts{expr->inputs().begin(),
+                                     expr->inputs().end()};
+  // TODO: Remove the temporary getenv
+  if (std::getenv("FUSER_SORT_INPUT_EXPRS") && respect_compute_at) {
+    if (expr->output(0)->getValType().value() == ValType::TensorView) {
+      auto out = expr->output(0)->as<const TensorView>();
+      // Move input TVs that are computed at this expression backward
+      // so that they are visited later. If multiple inputs are
+      // computed at, move TVs that are computed at an inner loop nest
+      // further backward.
+      std::sort(
+          next_stmts.begin(),
+          next_stmts.end(),
+          [out](const Statement* i0, const Statement* i1) {
+            std::array<const Statement*, 2> inputs{i0, i1};
+            std::array<int, 2> compute_at_axes{-1, -1};
+            for (int i = 0; i < 2; ++i) {
+              if (inputs[i]->getValType().value() == ValType::TensorView) {
+                auto tv = inputs[i]->as<const TensorView>();
```

Comment:
nit: since inputs is an array of const Statement*, we can simply say inputs[i]->as<TensorView>(). It will still return a const TensorView* (the const is not needed)

Reply:
Thanks. Didn't know that. Updated.